events_task/test/my_app_web/controllers/auth_controller_test.exs

```diff
@@ -0,0 +1,98 @@
+defmodule MyApp.AuthControllerTest do
+  use MyAppWeb.ConnCase
+
+  @valid_email "sergei@gmail.com"
+  @invalid_email ""
+  @updated_email "sergeitrigubov@gmail.com"
+
+  describe "users" do
+    test "new login page" do
+      conn = get(build_conn(), auth_path(build_conn(), :new))
+      assert conn.path_info == ["login"]
+      assert conn.method == "GET"
+    end
+
+    test "fails create user with invalid params" do
+      conn = build_conn()
+
+      response =
+        conn
+        |> post(auth_path(conn, :create, %{"user" => %{"email" => @invalid_email}}))
+        |> response(200)
+
+      assert response =~ "can&#39;t be blank"
+    end
+
+    test "successfully create user with valid params" do
+      conn = build_conn()
+
+      response =
+        conn
+        |> post(auth_path(conn, :create, %{"user" => %{"email" => @valid_email}}))
+        |> response(200)
+
+      assert response =~ "Please check your email for a link to sign in."
+    end
+
+    test "authenticates successfully with valid magik link" do
+      conn = build_conn()
+      conn = post(conn, auth_path(conn, :create, %{"user" => %{"email" => @valid_email}}))
+      conn = get(conn, auth_path(conn, :callback, conn.assigns[:token]))
+      assert Guardian.Plug.current_resource(conn).email == @valid_email
+    end
+
+    test "fails authentication with invalid magik link" do
+      conn = build_conn()
+      conn = post(conn, auth_path(conn, :create, %{"user" => %{"email" => @valid_email}}))
+
+      conn =
+        get(
+          conn,
+          auth_path(
+            conn,
+            :callback,
+            "JzdWIiOiIzODg1IiwidHlwIjoibWFnaWMifQ.5SY7G8Jz7Dtkxvj6vhlWSmpBXUHwTFkpbxd3a4Z3ROmYKRD4ANLQ8yZ88JQNb6GHh7B1YFSNW8gpTzmFrcvxvw"
+          )
+        )
+
+      assert Guardian.Plug.current_resource(conn) == nil
+    end
+
+    test "lsuccessfully ogout" do
+      conn = build_conn()
+      conn = post(conn, auth_path(conn, :create, %{"user" => %{"email" => @valid_email}}))
+      conn = get(conn, auth_path(conn, :callback, conn.assigns[:token]))
+      assert Guardian.Plug.current_resource(conn) != nil
+      conn = get(conn, auth_path(conn, :destroy))
+      assert Guardian.Plug.current_resource(conn) == nil
+    end
+
+    test "successfully update user with valid params" do
+      conn = build_conn()
+
+      conn = post(conn, auth_path(conn, :create, %{"user" => %{"email" => @valid_email}}))
+      conn = get(conn, auth_path(conn, :callback, conn.assigns[:token]))
+      conn = post(conn, auth_path(conn, :update, %{"user" => %{"email" => "Updated23mail.ru"}}))
+
+      assert conn.private[:plug_session]["phoenix_flash"] == %{
```

Comment:
It's better to use helper functions from phoenix when possible ([get_flash/2](https://hexdocs.pm/phoenix/Phoenix.ConnTest.html#get_flash/2)) - phoenix guys won't break them on upgrade because they are part of public API. When you fetch stuff by hardcoded keys - nothing stops phoenix guys to rename/remove keys in next phoenix release.